### ISSUE TITLE
Daemon stops working after a while

### ISSUE THREAD

**Issue by luispabon** [yory8/clipman#8]
It looks like the daemon (run as `clipman -d` only) stops collecting anything from the clipboard after a while. I can bring up selector via rofi menu, and it does show previous clipboard contents, just not new ones after a certain point.

Killing & restarting the daemon seems to do the trick. I'm currently working around this by killing & respawning the daemon in a cronjob every hour.

I'm not entirely sure how to debug this though.

**Comment by yory8** [comment]
Run it like this: `clipman -d 1>> PATH/TO/LOGFILE 2>&1 &`. This will collect all logging to a file.

**Comment by luispabon** [comment]
@yory8 thank you, I'll do so and report back.

I compiled wl-clipboard from source just last week at the same time as I installed clipman via go get.

**Comment by luispabon** [comment]
nothing so far

**Comment by yory8** [comment]
did you disable your cronjob?

**Comment by luispabon** [comment]
Yep.

**Comment by luispabon** [comment]
```
2019/07/11 10:02:54 Error running wl-paste (demon.52): free(): double free detected in tcache 2
```

It's the only thing coming out of the logs after the bug re-occurred. Not much. I'm trying to get more.

**Comment by yory8** [comment]
Thank'you, I too got two double frees, which I'm currently investigating (waiting for the next time it happens). If those are the cause, they are not our bug, but wl-paste's. 

Are you able to find out what content you were copying when the double free occurred? Did it come from an uncommon application or something?

**Comment by yory8** [comment]
@luispabon I pushed a "debug" branch. If you run it instead than master, it will notify you when: 1) you get a double free 2) clipman exits. This should allow you to see if the bug is triggered by something you are doing.

**Comment by luispabon** [comment]
Just got this installed.

> Are you able to find out what content you were copying when the double free occurred? 

This got me thinking I've been messing around with grim and wl-copy and ppm captures instead of png. I'll report back.

**Comment by luispabon** [comment]
```
2019/07/15 10:32:24 error running wl-copy (demon.91): fork/exec /usr/bin/wl-copy: invalid argument
```
I get this from capturing my 3 outputs (one is 4k) into ppm and into wl-paste like so:
```
 grim -t ppm - | wl-copy
```

**Comment by luispabon** [comment]
On a smaller screenshot (to ppm):

```
~ grim -t ppm -g "$(slurp -b '#AFAFAFAF' -c '#FF3F3FAF' -s '#00000000' -w 3 -d)" - | wl-copy
```

```
2019/07/15 10:56:17 error running wl-copy (demon.91): fork/exec /usr/bin/wl-copy: argument list too long
```
The clipman daemon also dies after that

**Comment by luispabon** [comment]
Seems to happen when having PNG data in the clipboard

**Comment by yory8** [comment]
Thank'you, this is very helpful. So it's actually dying even after catching the error; this test case (I can reproduce it) gives me a way to debug it, so I'm positive this issue will be closed soon.

As for the invalid argument bug, it is caused by a change I recently pushed to fix an issue when copying from neovim. I'll probably end up reverting it, as this is more serious, and the neovim issue is probably them doing non-standard stuff. I'll see to it very soon.

**Comment by yory8** [comment]
Also, the "fork/exec /usr/bin/wl-copy: argument list too long" bug is not ours, simply wl-copy has a size limit for the input.

**Comment by yory8** [comment]
Fixed in 0.9.0 (you should stop using the "debug" branch, which doesn't contain the fix; I'll replace it soon with proper desktop notifications in master).
Now:
1. we only copy text content
2. we only die if there's a failure in writing the history on disk, never on a wl-clipboard error.

Thank'you for your help and patience!

**Comment by luispabon** [comment]
Been testing this up and works like a charm. Thank you for the hard work!
